File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
import math
def solution(n, words):
    answer = []
    wdict = {}
    last = ''
    for idx, val in enumerate(words):
        if idx ==0 or last[-1] == val[0]:
            print(last,val)
            if val not in wdict:
                wdict[val] = 1
            else:
                if (idx+1)%n == 0:
                    return [n,math.ceil((idx+1)/n)]
                else:
                    return [(idx+1)%n,math.ceil((idx+1)/n)]
        else:
            if (idx+1)%n == 0:
                return [n,math.ceil((idx+1)/n)]
            else:
                return [(idx+1)%n,math.ceil((idx+1)/n)]
        
        last = val
    return [0,0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <map>
using namespace std;

vector<int> solution(int n, vector<string> words) {
    vector<int> answer(2);
    map<string,int> m;
    string cur = "", pre = "";
    m[words[0]]++;
    for (int i = 1; i < words.size(); i++) { 
        cur = words[i];
        pre = words[i-1];
        std::cout << cur << endl <<  pre << endl;
        if (cur.front() == pre.back()) {
            m[cur]++;
            // for (auto c:m) {
            // std :: cout << c.first << " : " << c.second << endl;    
            // }
            // std :: cout << m[cur] << endl;
            if (m[cur] > 1){
                answer[0] = i%n + 1;
                answer[1] = i/n + 1;
                return answer;
                }
            }
        else {
            answer[0] = i%n + 1;
            answer[1] = i/n + 1;
            return answer;
            }
    }       
    return answer;
}